common/src/log.rs

```diff
@@ -54,8 +54,15 @@ pub struct LogItem {
     pub line: String,
 }
 
+const LOG_LINE_MAX_CHARS: usize = 2048;
+
 impl LogItem {
-    pub fn new(id: Uuid, internal_origin: Backend, line: String) -> Self {
+    pub fn new(id: Uuid, internal_origin: Backend, line: impl Into<String>) -> Self {
+        let mut line = line.into();
+        if line.chars().count() > LOG_LINE_MAX_CHARS {
```

Comment:
Unless I am reading this incorrectly we now iterate over all the chars each time we create a new LogItem. Should we rather check the `len()`, which is stored with the string?

Reply:
And if the line is over the limit, we still need to iterate over the entire string before we can know the count is over the limit.